### DISCUSSION TITLE
Per File auto "norg"

### DISCUSSION THREAD

**Discussion by alexjp** [nvim-neorg/neorg#864] [Ideas]
Hi,

First thank you for the awesome project. I just found it out recently, so please bear with me.

I would wonder if it would be possible and a good idea to have a per file "norg" file, meaning for example:

* I am editing a file, for example: main.ex
* I press a key and neovim opens a main.ex.norg in the same folder (right next to the main.ex file) (it autocreates if it doesn't exist)
* I press a shortcut and it toggles between main.ex and main.ex.norg file.

Maybe if a good idea, some cool automatations like "copy  this like of code to main.ex.norg code block" for example , or linkage/templates for file/project types...

What do you think?

Thank you!

**Comment by d-r-a-b** [comment]
The basics of this is something you can easily setup yourself. Read `:h nmap` `:h alternate-file` `:h ctrl-^`. 

In terms of integrating things to ship nicely with Neorg, I imagine someone will think about more robust templating systems eventually but at the moment the main devs seem occupied with more pressing architecture concerns. Neorg has a great module system; if you feel strongly about this idea and have a vision for it go ahead and make it real and re-usable. https://github.com/andreadev-it/neorg-module-tutorials is a great tutorial for now and there's lot of info in the Wiki.

**Reply by alexjp** [reply]
Thanks.

yeah, after writing this and thinking about it more, I imagined that maybe its something that just makes sense in my use case, and not something standard.

thank you for your time. If I manage to do something that works well and its flexible, i'll make it available!

EDIT:

I had no clue it was this simple and easy, thanks for the hints !
```
vim.keymap.set("n", "<C-,>", function()
  local filename = vim.fn.expand("%:p")
  if string.find(filename, ".norg") then
    vim.cmd("e " .. filename:gsub(".norg", ""))
  else
    vim.cmd("e " .. filename .. ".norg")
  end
end, { desc = "Toggle between norg and file" })
```
